test/CellarWithMultiAssetDeposit.t.sol

```diff
@@ -0,0 +1,331 @@
+
+        string memory cellarName = "Cellar V0.0";
+        uint256 initialDeposit = 1e6;
+        deal(address(USDC), address(this), initialDeposit);
+        USDC.approve(0xa0Cb889707d426A7A386870A03bc70d1b0697598, initialDeposit);
+        uint64 platformCut = 0.75e18;
+        cellar = new CellarWithMultiAssetDeposit(
+            address(this),
+            registry,
+            USDC,
+            cellarName,
+            "POG",
+            usdcPosition,
+            abi.encode(true),
+            initialDeposit,
+            platformCut,
+            type(uint192).max
+        );
+
+        // Set up remaining cellar positions.
+        cellar.addPositionToCatalogue(wethPosition);
+        cellar.addPosition(0, wethPosition, abi.encode(true), false);
+        cellar.addPositionToCatalogue(wbtcPosition);
+        cellar.addPositionToCatalogue(usdtPosition);
+        cellar.addPosition(0, usdtPosition, abi.encode(true), false);
+
+        cellar.setStrategistPayoutAddress(strategist);
+        vm.label(address(cellar), "cellar");
+        vm.label(strategist, "strategist");
+        // Approve cellar to spend all assets.
+        USDC.approve(address(cellar), type(uint256).max);
+        initialAssets = cellar.totalAssets();
+        initialShares = cellar.totalSupply();
+    }
+
+    // ========================================= HAPPY PATH TEST =========================================
+
+    // Can we accept the Curve LP tokens? I dont see why not.
+
+    function testDeposit(uint256 assets) external {
+        assets = bound(assets, 1e6, 1_000_000_000e6);
+
+        deal(address(USDC), address(this), assets);
+        USDC.safeApprove(address(cellar), assets);
+
+        cellar.deposit(assets, address(this));
+
+        assertEq(
+            cellar.totalAssets(),
+            initialAssets + assets,
+            "Cellar totalAssets should equal initial + new deposit."
+        );
+        assertEq(
+            cellar.totalSupply(),
+            initialAssets + assets,
+            "Cellar totalSupply should equal initial + new deposit."
+        ); // Because share price is 1:1.
+    }
+
+    function testDepositWithAlternativeAsset(uint256 assets) external {
+        assets = bound(assets, 1e6, 1_000_000_000e6);
+
+        // Setup Cellar to accept USDT deposits.
+        cellar.setAlternativeAssetData(USDT, usdtPosition, 0);
+
+        deal(address(USDT), address(this), assets);
+        USDT.safeApprove(address(cellar), assets);
+
+        bytes memory depositCallData = abi.encodeWithSelector(Cellar.deposit.selector, assets, address(this), USDT);
+
+        address(cellar).functionCall(depositCallData);
+
+        // Since share price is 1:1, and USDT is hardcoded to equal the same as USDC, below checks should pass.
+        assertEq(cellar.previewRedeem(1e6), 1e6, "Cellar share price should be 1.");
+
+        cellar.dropAlternativeAssetData(USDT);
```

Comment:
Should we remove this and the commented out LoC below? You test dropping an asset later on in a different test I believe.

Reply:
Yeah actually I am not sure why I drop it there, also why I kept that commented out LoC. I'll remove them.